pokemon-checker/src/components/PokemonDisplay/PokemonDisplay.tsx

```diff
@@ -20,82 +19,47 @@ type displayProps = {
   getPokemonData: (url: string) => Promise<any>;
 };
 
-type displayState = {
-  pokemonObject: {
-    name?: string;
-    id?: string;
-    moves?: string[];
-    type1?: string;
-    type2?: string | null;
-    sprites?: {
-      frontDefault: string;
-      frontShiny: string;
-    };
-  };
-};
-
-
-
-class PokemonDisplay extends React.Component<displayProps, displayState> {
-  private pokemonToDisplay: PokemonDTO | null;
+export const PokemonDisplay = (props: displayProps) => {
+  const [pokeonObject, setPokemonObject] = useState<PokemonDTO>();
   //TODO (jeremy): Move this factory to a service! Views shouldn't control this.
-  private pokemonFactory: PokemonFactory;
-  private abilityService: AbilityService;
-
-  constructor(props: displayProps) {
-    super(props);
-    this.state = {
-      pokemonObject: {},
-    };
-    this.pokemonToDisplay = null;
-    this.pokemonFactory = new PokemonFactory();
-    this.abilityService = new AbilityService();
-  }
+  const pokemonFactory: PokemonFactory = new PokemonFactory();
 
-  componentDidUpdate(prevProps: displayProps) {
-    const prevUrl = prevProps.pokemonUrl;
-    const url = this.props.pokemonUrl;
-    if (prevUrl !== url) {
-      this.fetchPokemonObject();
-    }
-  }
+  useEffect(() => {
+    fetchPokemonObject();
+  });
 
-  fetchPokemonObject() {
-    const url = this.props.pokemonUrl;
-    if (url && url !== "") {
-      this.props
+  const fetchPokemonObject = () => {
+    const url = props.pokemonUrl;
+    if (url && url.length > 0) {
```

Comment:
This was previously `(url && url !== "")`, but because strings have a `.length` method I felt like it may not have been robust enough.

Really small nitpick but it might save us down the line.